packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -323,6 +331,69 @@ export async function uploadData<T extends UploadDataLookup, K extends keyof T>(
   );
 }
 
+export function shrinkRawData<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+): Boolean {
+  let totalSize = Buffer.byteLength(JSON.stringify(data));
+
+  while (totalSize > UPLOAD_SIZE_MAX) {
+    let largestEntryKey = '';
+    let largestEntrySize = 0;
+    let largestRawDataEntryKey = '';
+    let largestRawDataEntrySize = 0;
+    let largestItemKey = '';
+    let largestItemSize = 0;
+
+    // Find largest Entity
+    for (const entry in data) {
+      if (JSON.stringify(data[entry]).length > largestEntrySize) {
+        largestEntrySize = JSON.stringify(data[entry]).length;
+        largestEntryKey = entry;
```

Comment:
I thought about and even started to perform the sorting, but decided to bank on the assumption that most of the time we'd only be removing one very large value and not typically need to go through the loop again.  In the case that we only need to perform one removal, simply finding the largest is quicker than sorting the entire thing.  I can certainly switch to the method of sorting that you'd mentioned though.

Reply:
I don't want to derail this in a different direction if it requires too much work / adds to much complexity, given @VDubber comment here and below, I was wondering why we don't try to split the batch into more pieces?

I'm not sure if data integrity is high priority for `rawData`, but if we scan the data with `JSON.stringyify(data[entry]).length` and find that every object is less than 6MB each, we could retain all the data by creating several batches of size < `MAX_SIZE`.

Otherwise we could send what will fit, and then apply a truncating strategy to the remaining objects >6MB. I'm not sure if this was discussed or if retaining all `_rawData` is a high priority, so would love to know your thoughts.

Reply:
I like the idea of making it robust enough to just work with any sized entity, but I also don't know that we'd be happy in the long run if we let our integrations potentially start uploading 10+ MB entities.  That could get fairly painful from a database/storage/cost perspective.  Truncating at 6 MB feels like we've still given entities plenty of storage.

I think I'd be more inclined to go the route of reducing batch sizes to succeed with all sizes if we were running into entities of 6+ MB in non-raw data, but even then that feels like a case where the integration likely needs to break that out into multiple entities to really gain the most value out of storing it in a graph database.